components/brave_wallet_ui/common/async/handlers.ts

```diff
@@ -101,35 +101,6 @@ async function updateAccountInfo (store: Store) {
   }
 }
 
-async function updateCoinAccountNetworkInfo (store: Store, coin: BraveWallet.CoinType) {
-  const { accounts } = getWalletState(store)
-  if (accounts.length === 0) {
-    return
-  }
-  const { braveWalletService, keyringService } = getAPIProxy()
-
-  // Update Selected Coin & cached selected network
-  await store
-    .dispatch(walletApi.endpoints.setSelectedCoin.initiate(coin))
-    .unwrap()
-
-  // Updated Selected Account
-  const { address: selectedAccountAddress } =
-    coin === BraveWallet.CoinType.FIL
-      ? await keyringService.getFilecoinSelectedAccount(
-          (
-            await braveWalletService.getChainIdForActiveOrigin(coin)
-          ).chainId
-        )
-      : await keyringService.getSelectedAccount(coin)
-
-  const defaultAccount =
-    accounts.find(
-      (account) => account.address === selectedAccountAddress
-    ) || accounts[0]
-  await store.dispatch(WalletActions.setSelectedAccount(defaultAccount))
-}
-
```

Comment:
No need to fix selected account now.
Caller just invalidates cache of selected chain